### PR TITLE
d&a Indecomposable(Projective/Injective)Objects

### PR DESCRIPTION
**Pull request by kamalsaleh**
and enable WrapperCategory to lift these methods.

### CODE REVIEW

File: CAP/gap/WrapperCategory.gi [hunk 1]
```diff
@@ -397,6 +397,26 @@ InstallMethod( WrapperCategory,
...
+    if "IndecomposableInjectiveObjects" in list_of_operations_to_install then
+        
+        AddIndecomposableInjectiveObjects( D,
+          function( cat )
+            
+            return List( IndecomposableInjectiveObjects( ModelingCategory( cat ) ), a -> ModeledObject( cat, a ) );
+            
+        end );
+    fi;
+
```
[the remark this answers is not in the document]
Comment (kamalsaleh) [reply CAP/gap/WrapperCategory.gi]: Of course, I've already force-pushed!

File: CAP/gap/CategoryConstructor.gi [hunk 2]
```diff
@@ -37,6 +37,7 @@ InstallMethod( CategoryConstructor,
         create_func_object_or_fail := IsObject,
         create_func_morphism := IsObject,
         create_func_morphism_or_fail := IsObject,
+        create_func_list_of_objects := IsList
```
Comment (zickgraf) [inline CAP/gap/CategoryConstructor.gi:40]:
```suggestion
        create_func_list_of_objects := IsList,
```

File: CAP/gap/CategoryConstructor.gi [hunk 3]
```diff
@@ -252,6 +253,16 @@ InstallMethod( CategoryConstructor,
...
+        list_of_objects := """
+            function( input_arguments... )
+              local underlying_result;
+              
+              underlying_result := operation_name( underlying_arguments... );
+              
+              return List( underlying_result, object -> top_object_getter( cat, object ) );
+              
+            end
+        """
```
Comment (zickgraf) [inline CAP/gap/CategoryConstructor.gi:265]:
```suggestion
        """,
```

File: CAP/gap/CategoryConstructor.gi [hunk 4]
```diff
@@ -371,7 +382,7 @@ InstallMethod( CategoryConstructor,
             
         fi;
         
-        if StartsWith( info.return_type, "object" ) then
+        if ForAny( [ "object", "list_of_objects" ], str -> StartsWith( info.return_type, str ) ) then
```
Comment (zickgraf) [inline CAP/gap/CategoryConstructor.gi:385]:
```suggestion
        if PositionSublist( info.return_type, "object" ) <> fail then
```

File: CAP/gap/WrapperCategory.gi [hunk 5]
```diff
@@ -221,6 +221,7 @@ InstallMethod( WrapperCategory,
         create_func_object_or_fail := "default",
         create_func_morphism := "default",
         create_func_morphism_or_fail := "default",
+        create_func_list_of_objects := "default"
```
Comment (zickgraf) [inline CAP/gap/WrapperCategory.gi:224]:
```suggestion
        create_func_list_of_objects := "default",
```

**Review by zickgraf — CHANGES_REQUESTED** [review]
Please also update the docu in `CategoryConstructor.gd`.

PS: If you are very motivated you can also update the dummy category and the opposite category :D

**Review by zickgraf — DISMISSED** [review]
Thanks!

File: CAP/gap/CategoryConstructor.gi [hunk 6]
```diff
@@ -37,6 +37,7 @@ InstallMethod( CategoryConstructor,
         create_func_object_or_fail := IsObject,
         create_func_morphism := IsObject,
         create_func_morphism_or_fail := IsObject,
+        create_func_list_of_objects := IsList,
```
Comment (zickgraf) [inline CAP/gap/CategoryConstructor.gi:40]:
```suggestion
        create_func_list_of_objects := IsObject,
```

This checks the type of the options. The option `create_func_list_of_objects` can be either a function or the string "default", so we generically check `IsObject`.

**Pull request merged**